File: android/sdk/ssdid-core/src/main/java/my/ssdid/sdk/api/NotificationsApi.kt
package my.ssdid.sdk.api

import my.ssdid.sdk.domain.model.Identity
import my.ssdid.sdk.domain.notify.NotifyManager

class NotificationsApi internal constructor(private val manager: NotifyManager) {
    suspend fun ensureInboxRegistered(): String? = manager.ensureInboxRegistered()
    suspend fun createMailbox(identity: Identity) = manager.createMailbox(identity)
    suspend fun deleteMailbox(identity: Identity) = manager.deleteMailbox(identity)
    suspend fun fetchAndDemux() = manager.fetchAndDemux()
    suspend fun ackPending(notificationId: String) = manager.ackPending(notificationId)
    suspend fun updateDeviceToken(platform: String, token: String) =
        manager.updateDeviceToken(platform, token)
}

File: sdk/ios/SsdidCore/Sources/SsdidCore/Api/NotificationsApi.swift
import Foundation

/// Facade for push notification and mailbox operations.
public struct NotificationsApi {
    let manager: NotifyManager

    @MainActor
    public func ensureInboxRegistered() async throws {
        try await manager.ensureInboxRegistered()
    }

    @MainActor
    public func createMailbox(identity: Identity) async throws {
        try await manager.createMailbox(for: identity)
    }

    @MainActor
    public func deleteMailbox(identity: Identity) async throws {
        try await manager.deleteMailbox(for: identity)
    }

    @MainActor
    public func fetchAndDemux() async throws {
        try await manager.fetchAndDemux()
    }

    @MainActor
    public func ackPending(notificationId: String) async throws {
        try await manager.ackPending(notificationId: notificationId)
    }
}
